### PR TITLE
Plugin should have secure defaults

### PR DESCRIPTION
**Pull request by mreichel**
#427 Plugin should have secure defaults

### CODE REVIEW

**Comment by omehegan** [comment]
I just want to test this manually to confirm it's good before we merge.

File: src/main/java/com/dabsquared/gitlabjenkins/connection/GitLabConnectionConfig.java [hunk 1]
```diff
@@ -55,11 +55,11 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
         return super.configure(req, json);
     }
 
-    public boolean isUseAuthenticatedEndpoint() {
+    public Boolean getUseAuthenticatedEndpoint() {
```
Comment (christ66) [inline src/main/java/com/dabsquared/gitlabjenkins/connection/GitLabConnectionConfig.java]: I am 👎 as this causes binary incompatibility. This should be `isUseAuthenticatedEndpoint` as this is a normal java convention for boolean properties`is<propertyName>(){....}`

**Comment by christ66** [comment]
Other than the minor change, 👍

**Comment by mreichel** [comment]
I changed the name, because otherwise it did not trigger.

**Comment by christ66** [comment]
It might not work because you're using `Boolean` instead of `boolean` which it does not recognize the object version of boolean. I would recommend switching back to `boolean` as using the `Boolean` adds no value and it is possible to invoke `setUseAuthenticatedEndpoint(null)` which should not be possible.

**Comment by mreichel** [comment]
Using the wrapper type is the idea of the commit because we need the null value (in [#427](https://github.com/jenkinsci/gitlab-plugin/issues/427) @daniel-beck is giving details on the original idea)

**Comment by daniel-beck** [comment]
Right, but the API does not need changing; the API can remain `boolean`, you just need to make sure to have no paths that keep the `null` value between constructor and `readResolve` (and if you do it right, you won't).

**Comment by mreichel** [comment]
Ok, I made the changes we discussed

**Comment by daniel-beck** [comment]
This looks correct; but with something like this I recommend performing various upgrade tests.

**Comment by omehegan** [comment]
Yes, I'm intending to set aside time to do some manual upgrade testing to make sure we don't mess anything up.

**Review by christ66 — APPROVED** [review]
Agree with the testing. Other than that it LGTM.

**Comment by omehegan** [comment]
OK, I have tested this in the Docker setup that is included with the plugin. My process was:
1. Install 1.4.0 of the plugin and configure it to talk to GitLab. It defaults to no auth for the plugin endpoint.
2. Installed the current version of the plugin with this patch. Auth is still not enabled.
3. Uninstalled the plugin and manually cleared its config from JENKINS_HOME.
4. Installed the current version with patch again. Now auth shows up as enabled by default.

I pronounce this ready to merge. Hope we don't break anything!

**Comment by daniel-beck** [comment]
🎉

**Comment by omehegan** [comment]
This was released today in version 1.4.8 of the plugin.

**Pull request merged**